cppcache/src/PdxHelper.cpp

```diff
@@ -137,28 +136,18 @@ void PdxHelper::serializePdx(
     // so we don't know whether user has used those or not;; Can we do some
     // trick here?
 
-    auto createPdxRemoteWriter = [&]() -> PdxRemoteWriter {
-      if (auto pd = pdxTypeRegistry->getPreserveData(pdxObject)) {
-        auto mergedPdxType = pdxTypeRegistry->getPdxType(pd->getMergedTypeId());
-        return PdxRemoteWriter(output, mergedPdxType, pd, pdxTypeRegistry);
-      } else {
-        return PdxRemoteWriter(output, className, pdxTypeRegistry);
-      }
-    };
+    auto prw = cacheImpl->getPdxRemoteWriterFactory().create(
```

Comment:
What this extracted into an pure virtual interface for the sole purpose of integration testing?

Reply:
Exactly, I needed to force the race-condition and this was the only way I came up with. I am not sure there is any other way to force the race-condition.